### ISSUE TITLE
[bug]: GraphQL query to get categories returns categories that are "disabled" in admin

### ISSUE THREAD

**Issue by luke-denton-aligent** [magento/magento2#27127]
I have been trying to request all categories, categories' children, and categories' children's children (etc.) for my progressive web app, but the application is crashing when it tries to read `children_count` of the category in `useCategoryTree` (a talon in peregrine).

The issue appears to be that disabled categories that have children that _aren't_ disabled, are still returned from the query.

**Expected behavior**
(I think) The returned data should only return the top level categories that are enabled in admin, regardless of whether their children are also enabled.
If the intention _is_ to load top level categories when their children are active, then the data returned shouldn't be `null` for that top level category

### Preconditions (*)

1. Magento 2.3.3
2. PHP 7.3.11

### Steps to reproduce (*)

1.
2.

### Expected result (*)
1. Go to Categories in the Admin
2. Disable one of the categories
3. Ensure that disabled category has a child category that isn't disabled
4. Run the GraphQL query to get all categories, including children and children of children, with `"id": 1`:
```
query navigationMenu($id: Int!) {
    category(id: $id) {
        id
        name
        children {
            children_count
            id
            include_in_menu
            name
            children {
                name
                id
                include_in_menu
              	children_count
                children {
                    name
                    id
                    include_in_menu
                    children_count
                }
            }
        }
    }
}
```
5. See that the returned data includes an entry with data similar to the following
```
{
  "name": null,
  "id": 29,
  "include_in_menu": null,
  "children_count": null,
  "children": [ ... ]
```

### Actual result (*)
1. The category structure in my admin (it's the Luma sample data)
![image](https://user-images.githubusercontent.com/18253422/75635469-31ca9100-5c66-11ea-9a0c-59a31e69cde1.png)
2. GraphQL query and response in Graphiql
![image](https://user-images.githubusercontent.com/18253422/75635475-460e8e00-5c66-11ea-96fd-fee7a53689ad.png)
3. Behaviour when the same category is then enabled:
4. Admin category structure
![image](https://user-images.githubusercontent.com/18253422/75636038-24fc6c00-5c6b-11ea-8fae-f92a179a89e2.png)
54. GraphQL query and response in Graphiql
![image](https://user-images.githubusercontent.com/18253422/75636108-e3b88c00-5c6b-11ea-9672-c831cde50343.png)

(Originally posted to the [PWA Studio repository here](https://github.com/magento/pwa-studio/issues/2203), but have moved to this main repository, as it relates to the server side code, rather than client side)

**Possible solutions**
The `category` GraphQL query could be updated to allow us to specify an `include_in_menu` value in the query parameters

**Comment by engcom-Bravo** [comment]
Hello @luke-denton-aligent 

Thank you for your report and detailed description.

This seems to be fixed in Magento 2.4-develop.
We have rechecked it with Sample Data and disabled _Promotions_ category
![image](https://user-images.githubusercontent.com/51680745/75969815-27afc900-5ed8-11ea-947b-97f7fbf353a5.png)

If the Category is disabled, then the subcategories are not "returned" in the GraphQL query. Please, see the result
```graphql
{
  "data": {
    "category": {
      "id": 1,
      "name": "Root Catalog",
      "children": [
        {
          "children_count": "39",
          "id": 2,
          "include_in_menu": 1,
          "name": "Default Category",
          "children": [
            {
              "name": "Gear",
              "id": 6,
              "include_in_menu": 1,
              "children_count": "3",
              "children": [
                {
                  "name": "Bags",
                  "id": 7,
                  "include_in_menu": 1,
                  "children_count": "0"
                },
                {
                  "name": "Fitness Equipment",
                  "id": 8,
                  "include_in_menu": 1,
                  "children_count": "0"
                },
                {
                  "name": "Watches",
                  "id": 9,
                  "include_in_menu": 1,
                  "children_count": "0"
                }
              ]
            },
            {
              "name": "Men",
              "id": 14,
              "include_in_menu": 1,
              "children_count": "8",
              "children": [
                {
                  "name": "Tops",
                  "id": 15,
                  "include_in_menu": 1,
                  "children_count": "4"
                },
                {
                  "name": "Bottoms",
                  "id": 16,
                  "include_in_menu": 1,
                  "children_count": "2"
                }
              ]
            },
            {
              "name": "Training",
              "id": 12,
              "include_in_menu": 1,
              "children_count": "1",
              "children": [
                {
                  "name": "Video Download",
                  "id": 13,
                  "include_in_menu": 1,
                  "children_count": "0"
                }
              ]
            },
            {
              "name": "Women",
              "id": 23,
              "include_in_menu": 1,
              "children_count": "8",
              "children": [
                {
                  "name": "Tops",
                  "id": 24,
                  "include_in_menu": 1,
                  "children_count": "4"
                },
                {
                  "name": "Bottoms",
                  "id": 25,
                  "include_in_menu": 1,
                  "children_count": "2"
                }
              ]
            },
            {
              "name": "What's New",
              "id": 41,
              "include_in_menu": 1,
              "children_count": "0",
              "children": []
            },
            {
              "name": "Category",
              "id": 3,
              "include_in_menu": 1,
              "children_count": "0",
              "children": []
            },
            {
              "name": "Sale",
              "id": 40,
              "include_in_menu": 1,
              "children_count": "0",
              "children": []
            }
          ]
        },
        {
          "children_count": "1",
          "id": 4,
          "include_in_menu": 1,
          "name": "Web2",
          "children": [
            {
              "name": "Cat2",
              "id": 5,
              "include_in_menu": 1,
              "children_count": "0",
              "children": []
            }
          ]
        }
      ]
    }
  }
}
```
As you can see, there is Promotion category and the Promotions subcategories

You might also consider using the following query on 2.4-develop
```graphql
query navigationMenu {
  categoryList {
    id
    name
    children {
      children_count
      id
      include_in_menu
      name
      children {
        name
        id
        include_in_menu
        children_count
        children {
          name
          id
          include_in_menu
          children_count
        }
      }
    }
  }
}
```

Can you please confirm that you are not facing this issue on Magento 2.4-develop?

Thank you in advance

**Comment by luke-denton-aligent** [comment]
Hi @engcom-Bravo, thanks for taking a look at this. I'm using a Magento Cloud instance as the BE, I don't think I have the ability to change the version easily, so I won't be able to test this. Is there an ETA for the release of 2.4?

Thanks

**Comment by engcom-Bravo** [comment]
Hello @luke-denton-aligent 

Thank you for the quick response.

To get a Magento 2.4-develop instance, you simply need to add a comment with the following text to this issue 
`@magento give me 2.4-develop instance` - upcoming 2.4.x release
This will deploy a test vanilla instance with a Link, Login and Password. Like on this example below
![give_me_insrtance](https://user-images.githubusercontent.com/51680745/76061133-8e41ef00-5f8b-11ea-86a5-a0a2c63720d4.png)

The next release version, which should contain the fix for your issue, will be released in April 

Please, confirm if you could reproduce this issue on the deployed instance

Thank you in advance

**Comment by magento-engcom-team** [comment]
Hi @engcom-Bravo. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @engcom-Bravo, here is your [Magento instance](https://i-27127-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-27127-2-4-develop.instances.magento-community.engineering/admin_e6af](https://i-27127-2-4-develop.instances.magento-community.engineering/admin_e6af)
Login: `64d54e69` Password: `68edbb2f7dcb`
Instance will be terminated in up to 3 hours.

**Comment by luke-denton-aligent** [comment]
@engcom-Bravo wow that's really cool! Thanks for spinning that up, but either the instance has terminated itself, or I don't have access, as I get a 403 when trying to go to the instance URL.

I will give this tag a go though: @magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @luke-denton-aligent. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @luke-denton-aligent, here is your [Magento instance](https://i-27127-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-27127-2-4-develop.instances.magento-community.engineering/admin_2707](https://i-27127-2-4-develop.instances.magento-community.engineering/admin_2707)
Login: `86f1624b` Password: `ab832c61ec3f`
Instance will be terminated in up to 3 hours.

**Comment by luke-denton-aligent** [comment]
@engcom-Bravo, the new instance has just deployed and I was able to confirm that the issue _is_ fixed in 2.4.
I created the following category structure:
```
Default
|-- Category 1
|    |-- Subcategory 1-1
|    |-- Subcategory 1-2
|-- Category 2
|    |-- Subcategory 2-1
|    |-- Subcategory 2-2
|-- Category 3
|    |-- Subcategory 3-1
|    |-- Subcategory 3-2
```
And I disabled Category 2 _only_ (i.e. I left `Subcategory 2-1` and `Subcategory 2-2` enabled).
The result is that Category 2, and its children, were all left out of the response.

![image](https://user-images.githubusercontent.com/18253422/76261612-4f24de00-62aa-11ea-868a-21bee897b731.png)

There was one interesting thing about the returned results, as we can see above, Category 3 is before Category 1 in the results, so not sure if there is a sorting issue?

Other than sorting, the main concern that this Github Issue is reporting does appear to be fixed, so happy for it to be marked as resolved

**Comment by engcom-Bravo** [comment]
Hello @luke-denton-aligent 

That is great news. Thank you for your quick response.
So I will close this issue as not reproducible on 2.4-develop
Please feel free to report the _"Category 3 is before Category 1"_ and we will investigate it, in scope of another issue.

Thank you for collaboration
